frontend/public/components/monitoring/alert-manager-utils.tsx

```diff
@@ -36,6 +43,179 @@ export const patchAlertManagerConfig = (
 };
 
 export const receiverTypes = {
-  pagerduty: 'PagerDuty',
-  webhook: 'Webhook Receiver',
+  pagerduty_configs: 'PagerDuty',
+  webhook_configs: 'Webhook Receiver',
+};
+
+export const isKnownReceiverType = (receiverType: string) => {
+  return _.hasIn(receiverTypes, `${receiverType}_configs`);
+};
+
+/**
+ * Converts route object:
+ * {
+ *   receiver: "MyReceiver",
+ *   match: {
+ *     severity: "warning",
+ *     cluster: "myCluster"
+ *   },
+ *   match_re: {
+ *    service: "$foobar"
+ *  }
+};
+ * ...to array of labels for Routing Labels form component
+ * [
+ *   {
+ *     "name": "severity",
+ *     "value": "warning",
+ *     "isRegex": false
+ *   },
+ *   {
+ *     "name": "cluster",
+ *     "value": "myCluster",
+ *     "isRegex": false
+ *   },
+ *   {
+ *     "name": "service",
+ *     "value": "$foobar",
+ *     "isRegex": true
+ *   }
+ * ]
+ */
+export const getRouteLabelsForForm = (
+  receiver: AlertManagerReceiver,
+  routes: AlertManagerRoute[],
+) => {
+  const routesOfReceiver = _.find(
+    routes,
+    (aRoute: AlertManagerRoute) => aRoute.receiver === receiver.name,
+  );
+  const matches = _.map(_.get(routesOfReceiver, 'match', {}), (v, k) => {
+    return { name: k, value: v, isRegex: false };
+  });
+  const regexMatches = _.map(_.get(routesOfReceiver, 'match_re', {}), (v, k) => {
+    return { name: k, value: v, isRegex: true };
+  });
+  return _.concat([], matches, regexMatches);
+};
+
+/**
+ * Converts array of labels (from Form):
+ * [
+ *   {
+ *     "name": "severity",
+ *     "value": "warning",
+ *     "isRegex": false
+ *   },
+ *   {
+ *     "name": "cluster",
+ *     "value": "myCluster",
+ *     "isRegex": false
+ *   }
+ * ]
+ * ...to single label object, for Patch:
+ * {
+ *   "severity": "warning",
+ *   "cluster": "myCluster"
+ * }
+ */
+export const getRouteLabelsForPatch = (routeLabels, returnRegexMatches = false) => {
+  const matches = _.filter(routeLabels, (routeLabel: any) =>
+    returnRegexMatches ? routeLabel.isRegex : !routeLabel.isRegex,
+  );
+  const matchObj = _.map(matches, (label: any) => {
+    return { [label.name]: label.value };
+  });
+  return _.assign({}, ...matchObj);
+};
+
+/**
+ * Returns new Route object
+ * Ex:
+ * {
+ *   receiver: myNewReceiver,
+ *   match: {
+ *     "severity": "warning",
+ *     "cluster": "myCluster"
+ *   }
+ *   match_re {
+ *     "service": "^(foo1|foo2|baz)$",
+ *   }
+ * }
+ */
+export const createRoute = (receiver: AlertManagerReceiver, routeLabels) => {
+  const matchLabels = getRouteLabelsForPatch(routeLabels);
+  const matchReLabels = getRouteLabelsForPatch(routeLabels, true);
+
+  // construct new route object
+  const newRoute = { receiver: receiver.name };
+  if (!_.isEmpty(matchLabels)) {
+    _.set(newRoute, 'match', matchLabels);
+  }
+  if (!_.isEmpty(matchReLabels)) {
+    _.set(newRoute, 'match_re', matchReLabels);
+  }
+  return newRoute;
+};
```

Comment:
Think you can simplify this to something like
```
export const createRoute = (receiver: AlertManagerReceiver, routeLabels) => {
  const newRoute = { receiver: receiver.name };
  _.each(routeLabels, (label) => {
    _.set(newRoute, [label.isRegex ? 'match_re' : 'match', label.name], label.value);
  });
  return newRoute;
};
```

which means that `getRouteLabelsForPatch` is no longer needed.

Reply:
Updated `createRoute`.  `getRouteLabelsForForm` is still being used to set initial values in the BaseForm.